### PR TITLE
Evaluate file structures

### PR DESCRIPTION
**Pull request by davidleonfdez**
This PR adds a notebook and the auxiliary Python packages needed to compare different formats of the input files used to train the poems generator.

To compare the different formats without the bias that the extra content adds, I implemented a new loss that ignores the metadata added to the poems.

As suggested in a previous PR, I also added a doc folder with some guidelines and considerations about how to train a model with Hugging Face utilities.

From what I've tested there's isn't any significant difference between the formats. However, I've realised that, with any format, only the training loss decreases after the first 4 or 5 epochs. As I see it, this suggests that it could be a good idea to deploy two models:
* For the live performances of the poet, it's ok to have a model overfit to the training data, as the poems used there are the same as those used to train.
* For the web, however, it could be interesting to have a more regularized model, that performs better when it receives unseen poems or verses.

cc @ismaelfaro

### CODE REVIEW

File: doc/huggingface_hub.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+(remember that `model_url` should have the format "https://huggingface.co/{user}/{model_name}")
+
+However, if you want to save a model in the middle of a training run, in order to be able to continue later, during a different session, you should save the optimizer state too. The easiest way to do it is
+to call our method `nb_utils.commit_checkpoint_to_hf_hub`; of course, this will also push the tokenizer and the model. For instance:
+
+```
+commit_checkpoint_to_hub('gpt2-poems.en', 'davidlf', get_last_checkpoint('./checkpoints'),
+                          message='Save checkpoint at 10 epochs', pwd='[your_huggingface_password]')
```
Comment (ismaelfaro) [inline doc/huggingface_hub.md]: We should be generic in the demos I mean, instance use any username we should set the "placeholder" with something like "YOUR_HF_USER"

File: doc/huggingface_hub.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+To resume a training run from a checkpoint you stored in the Hub:
+
+1) Clone the repo:
+
+```
+custom_model_name = 'gpt2-poems.en'
+HF_USER = 'davidlf'
+hf_pwd = 'your_hf_password'
```
Comment (ismaelfaro) [inline doc/huggingface_hub.md]: same here:
> HF_USER = 'YOUR_HF_USER'
> hf_pwd = 'YOUR_HF_PASSWORD'

File: doc/training_considerations.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+# How to proceed
+
+In conclusion, the lesson to learn from all the above is that you should either:
+- a) Since the beginning, set `TrainingArguments.num_train_epochs` to the total number of training epochs you are running.
+- b) If you don't know beforehand the number of epochs you are going to train the model for, you can:
+  - Use a constant learning rate scheduler. For this, set the `lr_scheduler_type` param of `TrainingArguments` to
+   `SchedulerType.CONSTANT` or `SchedulerType.CONSTANT_WITH_WARMUP`. In the latter case, don't 
+    forget to set also the parameter `warmup_steps`.
+  - Pass your own scheduler to Trainer.\_\_init\_\_. The parameter `optimizer` is a size 2 tuple, with the optimizer being the first element and the scheduler, the second one.
```
Comment (ismaelfaro) [inline doc/training_considerations.md:75]: I think that this document can be converted In a useful Blog post ;)

**Review by ismaelfaro — COMMENTED** [review]
good set of tools, I'll check the code

File: doc/huggingface_hub.md (hunk 1 above)
Comment (davidleonfdez) [reply doc/huggingface_hub.md]: Oops! You are right, I'll check it everywhere.
